nav2_map_server/src/map_io.cpp

```diff
@@ -504,11 +504,14 @@ void tryWriteMapToFile(
     tf2::Matrix3x3 mat(tf2::Quaternion(orientation.x, orientation.y, orientation.z, orientation.w));
     double yaw, pitch, roll;
     mat.getEulerYPR(yaw, pitch, roll);
+    
+    int file_name_index = mapdatafile.find_last_of("/\\");
```

Comment:
Please change `int` -> `const int` to intentionally tell the compiler that this variable could be optimized out